alf/trainers/off_policy_trainer.py

```diff
@@ -124,21 +132,37 @@ def train_():
         env.reset()
         time_step = driver.get_initial_time_step()
         policy_state = driver.get_initial_state()
+
+        collect_steps = initial_collect_steps
+        if not collect_steps:
+            collect_steps = num_steps_per_iter
+
+        time_step, policy_state = driver.run(
+            max_num_steps=collect_steps,
+            time_step=time_step,
+            policy_state=policy_state)
+
         for iter in range(num_iterations):
             t0 = time.time()
-            time_step, policy_state = driver.run(
-                max_num_steps=num_steps_per_iter,
-                time_step=time_step,
-                policy_state=policy_state)
 
-            experience = replay_buffer.gather_all()
+            if clear_replay_buffer:
+                experience = replay_buffer.gather_all()
+                replay_buffer.clear()
+            else:
+                experience, _ = replay_buffer.get_next(
```

Comment:
Do we might want to make replay_buffer an optional input rather than always initializing by driver.add_replay_buffer()? I am considering adding a sampler to current TF replay buffer, maybe one easy way without changing TF replay buffer is to use as_dataset() of the TF.replay_buffer, then it would be helpful if we could have some flexibility when fetching experience from the replay_buffer.

Reply:
perhaps we should do some modification to make it flexible in a separate change .